elastic/kibana
# [Beats CM] Add Heartbeat

### Summary

Add support for Heartbeat Monitors to Beats Centralized Management.

There are two features we want to add:
- Heartbeat Monitors: A targeted configuration block style for HB monitors; this is the most common use case, but does not allow for the use of more advanced features
- ~~Heartbeat Other: A less specific block type that allows for more varied configuration of the beat.~~

This PR will require https://github.com/elastic/beats/pull/9254 to be merged before Heartbeat can actually use the CM features we are adding here.

### Testing this PR
Make sure you can:
- Enroll a Heartbeat
- Define a monitor on a tag, associate to your HB
- Define an ES output on a tag, associate your HB
- Add additional monitors and make sure they're tracked as well by HB
  - Add a TCP and ICMP check if possible to make sure those work as well

### Checklist


### For maintainers
